module/polls/helper.js

```diff
@@ -0,0 +1,631 @@
+                        let i = 1;
+                        question.options.forEach( option => {
+                            options.push(
+                               {
+                                 value: "R" + i,
+                                 label: option.label,
+                                 unicode: option.value
+                               }
+                           )
+                           
+                            result[question.qid]["R" + i] = 0;
+                           
+                           ++i
+                        })
+                    }
+
+                    question.options = options;
+                    questionArray.push(question);
+                });
+                
+                pollDocument.questions = questionArray;
+                pollDocument.result = result;
+                
+                let createPollResult = await database.models.polls.create(pollDocument)
+
+                let link = await gen.utils.md5Hash(userId + "###" + createPollResult._id);
+
+                await database.models.polls.updateOne
+                (
+                    { _id : createPollResult._id },
+                    {
+                        $set : { link : link}
+                    }
+                )
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLL_CREATED,
+                    data: {
+                        link : appsPortalBaseUrl + appName + messageConstants.common.TAKE_POLL + link
+                    }
+                });
+                
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+    /**
+    * List all the polls.
+    * @method
+    * @name list
+    * @param {String} userId - userId
+    * @returns {JSON} - Polls list.
+    */
+
+    static list(userId= "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if(userId == ""){
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK);
+                }
+
+                let pollsList = await this.pollDocuments
+                (
+                    {
+                        createdBy: userId,
+                        isDeleted: false
+                    },
+                    [
+                        "name"
+                    ],
+                    {createdAt: 1}
+                )
+
+                if (!pollsList.length) {
+                    throw new Error(messageConstants.apiResponses.POLL_NOT_FOUND)
+                }
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLLS_LIST_FETCHED,
+                    data: pollsList
```

Comment:
@deepakr123 - Data should be returned as false if no data found.

Reply:
resolved